src/utils/daemon.rs

```diff
@@ -12,41 +12,59 @@ use chrono::{Timelike, Utc};
 use failure::Error;
 use log::{debug, error, info};
 use std::sync::Arc;
-use std::thread;
-use std::time::Duration;
+use tokio::{
+    runtime::Handle,
+    sync::Mutex,
+    task::{self, JoinHandle},
+    time::{self, Duration, Instant},
+};
 
-fn start_registry_watcher(
-    opts: DocBuilderOptions,
+async fn start_registry_watcher(
     pool: Pool,
     build_queue: Arc<BuildQueue>,
-) -> Result<(), Error> {
-    thread::Builder::new()
-        .name("registry index reader".to_string())
-        .spawn(move || {
-            // space this out to prevent it from clashing against the queue-builder thread on launch
-            thread::sleep(Duration::from_secs(30));
-            loop {
-                let mut doc_builder =
-                    DocBuilder::new(opts.clone(), pool.clone(), build_queue.clone());
-
-                if doc_builder.is_locked() {
-                    debug!("Lock file exists, skipping checking new crates");
-                } else {
-                    debug!("Checking new crates");
-                    match doc_builder.get_new_crates() {
-                        Ok(n) => debug!("{} crates added to queue", n),
-                        Err(e) => error!("Failed to get new crates: {}", e),
-                    }
-                }
+    options: DocBuilderOptions,
+) -> JoinHandle<()> {
+    task::spawn(async move {
+        let (cloned_pool, cloned_build_queue, cloned_options) =
+            (pool.clone(), build_queue.clone(), options.clone());
 
-                thread::sleep(Duration::from_secs(60));
-            }
-        })?;
+        let doc_builder = Arc::new(Mutex::new(
```

Comment:
This should be a `std::sync::Mutex` given that it is never locked concurrently (and use `.try_lock().unwrap()` instead of `.lock()` to ensure it never blocks).